tests.py

```diff
@@ -358,6 +366,52 @@ class TestBinaryModulatedWriter(TestWriter):
     label = "TestBinaryModulatedWriter"
     writer_class_to_test = pyfstat.BinaryModulatedWriter
 
+    def test_tp_parsing(self):
+        this_writer = self.writer_class_to_test(
+            outdir=self.outdir, **default_Writer_params, **default_binary_params,
+        )
+        this_writer.make_data()
+
+        theta_prior = {
+            key: value
+            for key, value in default_signal_params.items()
+            if key not in ["h0", "cosi"]
+        }
+        theta_prior.update({key: value for key, value in default_binary_params.items()})
+        theta_prior["tp"] = {
+            "type": "unif",
+            "lower": default_binary_params["tp"]
+            - 0.5 * default_binary_params["period"],
+            "upper": default_binary_params["tp"]
+            + 0.5 * default_binary_params["period"],
+        }
+        theta_prior.pop
+
+        mcmc_kwargs = {
+            "nsteps": [500],
```

Comment:
Have you checked if a shorter run may be sufficient? MCMC tests tend to be quite slow.

Reply:
This one runs fast on my computer, but I'll check again.